pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -60,6 +61,30 @@ func GetEnvironment(cr *api.KieApp, service kubernetes.PlatformService) (api.Env
 		cr.Status.Applied.Version = constants.CurrentVersion
 		cr.Spec.Version = ""
 	}
+
+	/*  with cr.Spec.CommonConfig.SecretAdminCredentialsReference != nil && len(cr.Spec.CommonConfig.AdminUser) >= 0 && len(cr.Spec.CommonConfig.AdminPassword) >= 0
+	we do nothing because is a condition used in the tests and we skip the secret check (not available in the unit tests)
+	*/
```

Comment:
Do nothing is not ideal, inmo. If we run to this state, maybe it would be good to remove secretAdmin form status aplied section in KieApp CRD, so in that case users would see that this config was not applied.
Also it would be good to mentioned it in description that credentials secret will be not used if adminUser and adminPassword is already set.